amethyst_renderer/src/sprite.rs

```diff
@@ -61,6 +68,145 @@ impl From<[f32; 4]> for Sprite {
+                position: [half_width, half_height, 0.0],
+                tex_coord: [1.0, 1.0],
+            },
+            PosTex {
+                position: [-half_width, half_height, 0.0],
+                tex_coord: [0.0, 1.0],
+            },
+            PosTex {
+                position: [half_width, -half_height, 0.0],
+                tex_coord: [1.0, 0.0],
+            },
+        ];
+
+        let mesh = self.loader
+            .load_from_data(vertices.into(), (), &self.asset_storage);
+
+        let material = Material {
+            albedo: texture,
+            albedo_offset: TextureOffset {
+                u: (sprite.left / size.0, sprite.right / size.0),
+                v: (1.0 - sprite.bottom / size.1, 1.0 - sprite.top / size.1),
+            },
+            ..self.material_defaults.0.clone()
+        };
+
+        (mesh, material)
+    }
+
+    /// Adds a mesh and a material to an entity corresponding to the sprite and texture given.
+    /// Note that is you need to insert the same sprite and texture, using ``add_sprite_render_multiple`` allows for better performances.
+    pub fn add_sprite_render(
+        &mut self,
+        entity: Entity,
+        sprite: Sprite,
+        texture: TextureHandle,
+        texture_size: (f32, f32),
+    ) -> Result<()> {
+        let (mesh, material) = self.build_mesh_and_material(sprite, texture, texture_size);
+        self.meshes.insert(entity, mesh)?;
+        self.materials.insert(entity, material)?;
+        Ok(())
+    }
+
+    /// Adds the same mesh and material to multiple entities corresponding to the sprite and texture given.
+    pub fn add_sprite_render_multiple(
+        &mut self,
+        entities: Vec<Entity>,
+        sprite: Sprite,
+        texture: TextureHandle,
+        texture_size: (f32, f32),
+    ) -> Result<()> {
+        let len = entities.len();
+        if len == 0 {
+            Err(Error::Arguments(
+                "0 entities were provided to add SpriteRender to".to_owned(),
+            ))
+        } else {
+            let (mesh, material) = self.build_mesh_and_material(sprite, texture, texture_size);
+            for entity in 0..len - 1 {
+                self.meshes.insert(entities[entity], mesh.clone())?;
+                self.materials.insert(entities[entity], material.clone())?;
+            }
+            self.meshes.insert(entities[len - 1], mesh)?;
+            self.materials.insert(entities[len - 1], material)?;
+            Ok(())
+        }
+    }
+}
+
+/// An easy way to attach and display a sprite when building an entity
+pub trait WithSpriteRender
+where
+    Self: Sized,
+{
+    /// Adds a mesh and a material to the entity being built corresponding to the sprite and texture given.
+    fn with_sprite_render(
+        self,
+        sprite: Sprite,
+        texture: TextureHandle,
+        texture_size: (f32, f32),
+    ) -> Result<Self>;
+}
+
+impl<'a> WithSpriteRender for EntityBuilder<'a> {
+    fn with_sprite_render(
```

Comment:
Isn't `with_sprite` enough? Also, please add an `add_sprite` variant.

Reply:
Well I assumed doing that would blur the line between the ``Sprite`` struct and the actual rendering process. Do you think it matters?

Reply:
Also, I believe that in the SpriteRenderData, the methods should be ``add`` and ``add_multiple`` instead.